mpi.c

```diff
@@ -39,31 +36,41 @@ int MPI_Finalize(void)
     return status;
 }
 
-// TODO: tags, generate a message envelope with match info so the recv side can match
 int MPI_Send(const void *buf, int count, MPI_Datatype datatype, int dest, int tag, MPI_Comm comm)
 {
     int status = MPI_SUCCESS;
 
+    char msg_envelope_buf[sizeof(nanompi_message_envelope) + (count * datatype.size )];
```

Comment:
It would be simpler and more performant to do a socket send for the envelope, then another socket send for the buffer itself. That saves a memory allocation and memcpy, which can be costly at larger message sizes